pkg/controller/node/status.go

```diff
@@ -60,24 +82,113 @@ func calculateStatus(cconfig *v1.ControllerConfig, pool *mcfgv1.MachineConfigPoo
 	}
 	machineCount := int32(len(nodes))
 
-	updatedMachines := getUpdatedMachines(pool, nodes)
-	updatedMachineCount := int32(len(updatedMachines))
+	/*
+		updatedMachines := getUpdatedMachines(pool.Spec.Configuration.Name, nodes)
+		updatedMachineCount := int32(len(updatedMachines))
 
-	readyMachines := getReadyMachines(pool, nodes)
-	readyMachineCount := int32(len(readyMachines))
+		readyMachines := getReadyMachines(pool.Spec.Configuration.Name, nodes)
+		readyMachineCount := int32(len(readyMachines))
+
+		unavailableMachines := getUnavailableMachines(nodes)
+		unavailableMachineCount := int32(len(unavailableMachines))
 
-	unavailableMachines := getUnavailableMachines(nodes, pool)
+		degradedMachines := getDegradedMachines(nodes)
+		degradedReasons := []string{}
+		for _, n := range degradedMachines {
+			reason, ok := n.Annotations[daemonconsts.MachineConfigDaemonReasonAnnotationKey]
+			if ok && reason != "" {
+				degradedReasons = append(degradedReasons, fmt.Sprintf("Node %s is reporting: %q", n.Name, reason))
+			}
+		}
+	*/
+	// in the event you are upgrading between versions, the statecontroller is going to get confused and it seems to depend on
+	// the operator pod being rolled out
+
+	var degradedMachines, readyMachines, updatedMachines, unavailableMachines, updatingMachines []*corev1.Node
+	// if we represent updating properly here, we will also represent updating properly in the CO
+	// so this solves the cordoning RFE and the upgradeable RFE
+	// updating == updatePrepared, updateExecuted, updatedComplete
+	// updated == nodeResumed, updated
+	// ready == nodeResumed, updated
+	// unavailable == draining, cordoned
+	// degraded == if the condition.Reason == error
+	// this ensures that a MCP only enters Upgradeable==False if the node actually needs to upgrade to the new MC
+	for _, state := range mcs {
+		var ourNode *corev1.Node
+		for _, n := range nodes {
+			if state.Name == n.Name {
+				ourNode = n
+				break
+			}
+		}
+		if ourNode == nil {
+			klog.Errorf("Could not find specificed node %s", state.Name)
+		}
+		if len(state.Status.Conditions) == 0 {
+			// not ready yet
+			break
+		}
+		for _, cond := range state.Status.Conditions {
+			if cond.Reason == "Error" {
+				degradedMachines = append(degradedMachines, ourNode)
+				continue
+			}
+			if cond.Status == metav1.ConditionUnknown {
```

Comment:
Hmm, why are all of these ConditionUnknown?

Reply:
explained on slack. but will re-iterate here. 

1. Unknown == progressing during most phases
2. False == Phase was skipped due to some config (drain for example gets skipped) or Phase has not happened yet 
3. True == we have completed this phase, onto the next!

If you notice, all of the API types are in the past tense: drained, cordoned, updateExecuted etc